enterprise_access/apps/content_assignments/management/commands/automatically_nudge_assignments.py

```diff
@@ -0,0 +1,135 @@
+from enterprise_access.apps.content_assignments.models import AssignmentConfiguration
+from enterprise_access.apps.content_assignments.utils import are_dates_matching_with_day_offset
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Automatically nudge learners who are 'days_before_course_start_date' days away from the course start_date
+    The default notification lead time without a provided `days_before_course_start_date` argument is 30 days
+    """
+    help = (
+        'Spin off celery tasks to automatically send a braze email to '
+        'remind learners about an upcoming accepted assignment a certain number '
+        'of days in advanced determined by the "days_before_course_start_date" argument'
+    )
+
+    def add_arguments(self, parser):
+        """
+        Entry point to add arguments.
+        """
+        parser.add_argument(
+            '--dry-run',
+            action='store_true',
+            dest='dry_run',
+            default=False,
+            help='Dry Run, print log messages without spawning the celery tasks.',
+        )
+        parser.add_argument(
+            '--days_before_course_start_date',
+            action='store_true',
+            dest='days_before_course_start_date',
+            default=30,
+            help='The amount of days before the course start date to send a nudge email through braze',
+        )
+
+    @staticmethod
+    def to_datetime(value):
+        """
+        Return a datetime object of `value` if it is a str.
+        """
+        if isinstance(value, str):
+            return datetime.datetime.strptime(
+                value,
+                "%Y-%m-%dT%H:%M:%SZ"
+            ).replace(
+                tzinfo=datetime.timezone.utc
+            )
+
+        return value
+
+    def handle(self, *args, **options):
+        dry_run = options['dry_run']
+        days_before_course_start_date = options['days_before_course_start_date']
+
+        for assignment_configuration in AssignmentConfiguration.objects.filter(active=True):
+            subsidy_access_policy = assignment_configuration.subsidy_access_policy
+            enterprise_catalog_uuid = subsidy_access_policy.catalog_uuid
+
+            message = (
+                '[AUTOMATICALLY_REMIND_ACCEPTED_ASSIGNMENTS_1] Assignment Configuration. UUID: [%s], '
+                'Policy: [%s], Catalog: [%s], Enterprise: [%s], dry_run [%s]',
+            )
+            logger.info(
+                message,
+                assignment_configuration.uuid,
+                subsidy_access_policy.uuid,
+                enterprise_catalog_uuid,
+                assignment_configuration.enterprise_customer_uuid,
+                dry_run,
+            )
+
+            accepted_assignments = assignment_configuration.assignments.filter(
+                state=LearnerContentAssignmentStateChoices.ACCEPTED
```

Comment:
You want to filter for only allocated assignments. `accepted` means they've already redeemed/enrolled.